### PR TITLE
logging: new mode -l passthrough-tty

### PR DESCRIPTION
**Pull request by giuseppe**
it works in a similar way to passthrough but it allows to be used also on a TTY.

conmon support: https://github.com/containers/conmon/pull/465

Closes: https://github.com/containers/podman/issues/20767

#### Does this PR introduce a user-facing change?

```release-note
Add a new logging mode --passthrough-tty that works like passthrough but can be used with TTYs
```

### CODE REVIEW

**Comment by rhatdan** [comment]
@giuseppe is this still a draft?

**Comment by halfline** [comment]
given `passthrough-tty` got folded into `passthrough` in the conmon merge request, should the same thing happen here?

**Comment by giuseppe** [comment]
> given `passthrough-tty` got folded into `passthrough` in the conmon merge request, should the same thing happen here?

personally I think it is better to keep it as a separate mode, to avoid potential security issues with `passthrough` mode.  Now it is explicit that the user expects the std streams to potentially be terminals and accept the risk

File: test/system/035-logs.bats
Comment (Luap99) [inline test/system/035-logs.bats:450]: This is no longer applicable I assume?

File: test/system/035-logs.bats
Comment (Luap99) [inline test/system/035-logs.bats:442]: looking at CI this seems to always trigger so it is unable to catch any regression in CI thus effectively making the test pointless. Can we setup a tty for this test somehow?

**Comment by edsantiago** [comment]
Hi, sorry I haven't been watching this closely, before you spend too much time on this, can you look at `450-interactive.bats`?

**Comment by giuseppe** [comment]
> Hi, sorry I haven't been watching this closely, before you spend too much time on this, can you look at `450-interactive.bats`?

thanks, that is surely better!

I've moved the test to `450-interactive.bats` and used the existing infrastructure

File: test/system/450-interactive.bats
Comment (edsantiago) [inline test/system/450-interactive.bats:111]: ummmmmm....

File: test/system/450-interactive.bats
Comment (edsantiago) [inline test/system/450-interactive.bats:117]: Do you know what version of conmon supports this, and where it is in the bodhi pipeline? I just spun up a rawhide VM, and even in updates-testing, I can't get a conmon that supports this. I would like to hold this PR until I can run this test against a truly working conmon.

File: test/system/450-interactive.bats
Comment (giuseppe) [reply test/system/450-interactive.bats]: dropped now

File: test/system/450-interactive.bats
Comment (giuseppe) [reply test/system/450-interactive.bats:113]: it is not released yet: https://github.com/containers/conmon/pull/465

Fine by me, let's hold it until the next conmon release

**Comment by edsantiago** [comment]
Ping. We have [new VMs](https://github.com/containers/automation_images/pull/320). Conmon is now `2.1.6+ds1-1` on Debian, and `2:2.1.8-2` on all Fedorae. Is that the required version?

**Comment by giuseppe** [comment]
I've rebased and cleaned up the test but unfortunately 2.1.8 is not enough.  We need at least 2.1.9 (which has a regression, so it should be 2.1.10).

File: docs/source/markdown/options/log-driver.md
Comment (TomSweeneyRedHat) [inline docs/source/markdown/options/log-driver.md:18]:
```suggestion
The **passthrough-tty** driver is the same as **passthrough** except that it also allows it to be used on a TTY if the user really wants it.
```

**Comment by rhatdan** [comment]
@giuseppe Could you rebase.

File: docs/source/markdown/options/log-driver.md
Comment (giuseppe) [reply docs/source/markdown/options/log-driver.md]: fixed

**Review by edsantiago — APPROVED** [review]

File: test/system/450-interactive.bats
Comment (giuseppe) [reply test/system/450-interactive.bats:108]: thanks, amended, it did the job!

**Comment by giuseppe** [comment]
@containers/podman-maintainers PTAL

**Pull request merged**
